torch/cuda/_memory_viz.py

```diff
@@ -752,9 +752,13 @@ def _output(p):
     def _read(name):
         if name == "-":
             f = sys.stdin.buffer
+            need_close = False
         else:
             f = open(name, "rb")
+            need_close = True
```

Comment:
Just do with f: context manager. If the stdin descript doesn't support it, use contextlib.closing wrapper